src/iron.rs

```diff
@@ -85,8 +108,13 @@ impl<H: Handler> Iron<H> {
 
 impl<H: Handler> ::hyper::server::Handler for Iron<H> {
     fn handle(&self, http_req: HttpRequest, http_res: HttpResponse<Fresh>) {
+        let protocol = match self.security {
+            Security::Insecure => "http",
+            Security::Secure { .. } => "https"
```

Comment:
This should be factored into a method and called later - `Request::from_http` should take a `Protocol/Security`.